src/layout/Number/NumberComponent.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+import { formatNumericText } from '@digdir/design-system-react';
+import cn from 'classnames';
+
+import { Lang } from 'src/features/language/Lang';
+import { useCurrentLanguage } from 'src/features/language/LanguageProvider';
+import { getMapToReactNumberConfig } from 'src/hooks/useMapToReactNumberConfig';
+import classes from 'src/layout/Number/NumberComponent.module.css';
+import type { PropsFromGenericComponent } from 'src/layout';
+
+export const NumberComponent = ({ node }: PropsFromGenericComponent<'Number'>) => {
+  const { textResourceBindings, value, icon, direction } = node.item;
+  const currentLanguage = useCurrentLanguage();
+
+  const getDisplaydata = (text) => {
+    const numberFormatting = getMapToReactNumberConfig(node.item.formatting, text, currentLanguage);
+    if (numberFormatting?.number) {
+      return formatNumericText(text.toString(), numberFormatting.number);
+    }
+    return text;
+  };
+
+  if (icon) {
+    const imgType = icon.split('.').at(-1);
+
+    if (!imgType) {
+      throw new Error('Image source is missing file type. Are you sure the image source is correct?');
+    }
+    if (!['svg', 'png', 'jpg', 'jpeg', 'gif', 'bmp', 'tiff'].includes(imgType.toLowerCase())) {
+      throw new Error('Only images of the types: .svg, .png, .jpg, .jpeg, .gif, .bmp, .tiff, are supported');
+    }
```

Comment:
I'm still not excited about these types of limitations - why do we need them? We're making a developer platform, so if our app developers configures something that's not valid as an image, that's on them - not us. I would love for apps to be able to use AVIF, WebP or any other new and fancy image format that gets into browsers in the future, without us having to add things to a whitelist.